llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp

```diff
@@ -4378,7 +4378,8 @@ void CodeGenDAGPatterns::ParsePatterns() {
       FindPatternInputsAndOutputs(Pattern, Pattern.getTree(j), InstInputs,
                                   InstResults, InstImpResults);
 
-    ParseOnePattern(CurPattern, Pattern, Result, InstImpResults);
+    if (!CurPattern->getValueAsBit("GISelShouldIgnore"))
```

Comment:
Wait, wouldn't that affect the DAGISel as well?
Can you add a DAGISel test to show that a GISelShouldIgnore pattern is still processed by the DAG and emitted?
Just take a basic test that checks the OPC_ opcodes emitted and add a run line with the GISelShouldIgnore bit enabled.

Reply:
Yes, that would affect it in this way, which isn't what we want... Reworking back to the prior commit.